src/main/java/teammates/common/util/TimeHelper.java

```diff
@@ -112,25 +117,44 @@ public static Calendar now(double timeZone) {
     }
 
     /**
-     * Convert a date string and time string into a Date object. Returns null on error.
-     *
+     * Convert a date string and time string of only the hour into a Date object.
+     * If the hour is 24, it is converted to 23:59. Returns null on error.
      * @param inputDate
      *            The date in format dd/MM/yyyy
      * @param inputTimeHours
-     *            The time as number of hours
+     *            0-24
      */
+    @Deprecated
     public static Date combineDateTime(String inputDate, String inputTimeHours) {
+        return convertLocalDateTimeToDate(combineDateTimeNew(inputDate, inputTimeHours));
+    }
+
+    /**
+     * Convert a date string and time string of only the hour into a Date object.
+     * If the hour is 24, it is converted to 23:59. Returns null on error.
+     * @param inputDate         the date string in dd/MM/yyyy format
+     * @param inputTimeHours    the hour, 0-24
+     * @return                  a LocalDateTime at the specified date and hour
+     */
+    public static LocalDateTime combineDateTimeNew(String inputDate, String inputTimeHours) {
         if (inputDate == null || inputTimeHours == null) {
             return null;
         }
 
-        int inputTimeInt = 0;
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy H.mm");
+        String dateTimeString;
+        if ("24".equals(inputTimeHours)) {
+            dateTimeString = inputDate + " 23.59";
+        } else {
+            dateTimeString = inputDate + " " + inputTimeHours + ".00";
+        }
         try {
-            inputTimeInt = Integer.parseInt(inputTimeHours) * 100;
-        } catch (NumberFormatException nfe) {
+            return LocalDateTime.parse(dateTimeString, formatter);
+        } catch (DateTimeParseException e) {
+            log.warning("Failed to parse datetime string: \""
+                    + dateTimeString + "\" with format " + formatter);
```

Comment:
Not sure we actually need to log this - if it happens, it's because the user sent custom POST parameters manually, which we are handling fine. The original code didn't log it after all.